### PR TITLE
[SYCLomatic] Support migration of thrust::set_symmetric_difference_by_key and thrust::set_symmetric_difference

### PR DESCRIPTION
**Pull request by tomflinda**
Signed-off-by: chenwei.sun <chenwei.sun@intel.com>

### CODE REVIEW

File: clang/test/dpct/thrust-algo-part2.cu [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+  thrust::set_symmetric_difference_by_key(thrust::host, A_keys, A_keys + 7, B_keys, B_keys + 5, A_vals, B_vals, keys_result, vals_result);
+  thrust::set_symmetric_difference_by_key(A_keys, A_keys + 7, B_keys, B_keys + 5, A_vals, B_vals, keys_result, vals_result);
+  thrust::set_symmetric_difference_by_key(thrust::host, A_keys, A_keys + 7, B_keys, B_keys + 5, A_vals, B_vals, keys_result, vals_result, Compare());
+  thrust::set_symmetric_difference_by_key(A_keys, A_keys + 7, B_keys, B_keys + 5, A_vals, B_vals, keys_result, vals_result, Compare());
+  thrust::set_symmetric_difference_by_key(thrust::device, d_A_keys.begin(), d_A_keys.end(), d_B_keys.begin(), d_B_keys.end(), d_A_vals.begin(), d_B_vals.begin(), d_keys_result.begin(), d_vals_result.begin());
+  thrust::set_symmetric_difference_by_key(d_A_keys.begin(), d_A_keys.end(), d_B_keys.begin(), d_B_keys.end(), d_A_vals.begin(), d_B_vals.begin(), d_keys_result.begin(), d_vals_result.begin());
+  thrust::set_symmetric_difference_by_key(thrust::device, d_A_keys.begin(), d_A_keys.end(), d_B_keys.begin(), d_B_keys.end(), d_A_vals.begin(), d_B_vals.begin(), d_keys_result.begin(), d_vals_result.begin(), Compare())
```
Comment (zhiweij1) [inline clang/test/dpct/thrust-algo-part2.cu]:
```suggestion
  thrust::set_symmetric_difference_by_key(thrust::device, d_A_keys.begin(), d_A_keys.end(), d_B_keys.begin(), d_B_keys.end(), d_A_vals.begin(), d_B_vals.begin(), d_keys_result.begin(), d_vals_result.begin(), Compare());
```

File: clang/test/dpct/thrust-algo-part2.cu [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+      thrust::set_symmetric_difference_by_key(d_A_keys.begin(), d_A_keys.end(), d_B_keys.begin(), d_B_keys.end(), d_A_vals.begin(), d_B_vals.begin(), d_keys_result.begin(), d_vals_result.begin(), Compare())
```
Comment (zhiweij1) [inline clang/test/dpct/thrust-algo-part2.cu]:
```suggestion
  thrust::set_symmetric_difference_by_key(d_A_keys.begin(), d_A_keys.end(), d_B_keys.begin(), d_B_keys.end(), d_A_vals.begin(), d_B_vals.begin(), d_keys_result.begin(), d_vals_result.begin(), Compare());
```

File: clang/test/dpct/thrust-algo-part2.cu [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+          thrust::set_symmetric_difference_by_key(thrust::host, h_A_keys.begin(), h_A_keys.end(), h_B_keys.begin(), h_B_keys.end(), h_A_vals.begin(), h_B_vals.begin(), h_keys_result.begin(), h_vals_result.begin());
```
Comment (zhiweij1) [inline clang/test/dpct/thrust-algo-part2.cu]:
```suggestion
  thrust::set_symmetric_difference_by_key(thrust::host, h_A_keys.begin(), h_A_keys.end(), h_B_keys.begin(), h_B_keys.end(), h_A_vals.begin(), h_B_vals.begin(), h_keys_result.begin(), h_vals_result.begin());
```

**Review by zhiweij1 — CHANGES_REQUESTED** [review]

File: clang/test/dpct/thrust-algo-part2.cu (hunk 1 above)
Comment (tomflinda) [reply clang/test/dpct/thrust-algo-part2.cu]: Updated

File: clang/test/dpct/thrust-algo-part2.cu (hunk 3 above)
Comment (tomflinda) [reply clang/test/dpct/thrust-algo-part2.cu]: Updated

**Review by zhiweij1 — APPROVED** [review]

**Review by zhimingwang36 — APPROVED** [review]
LGTM

**Pull request merged**